### PR TITLE
avoid constructor.name . fix for #65

### PR DESCRIPTION
**Pull request by ya7ya**
this avoids using `constructor.name` so uglify can mangle and compress `js-ipfs` properly.

this should close https://github.com/libp2p/js-libp2p/issues/65

cc @diasdavid

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -129,7 +130,7 @@ class Node extends EventEmitter {
         this.switch.transport.add(
           transport.tag || transport.constructor.name, transport)
       } else if (transport.constructor &&
-                 transport.constructor.name === 'WebSockets') {
+                 transport instanceof WebSockets) {
```
Comment (diasdavid) [inline src/index.js]: Same feedback as https://github.com/multiformats/js-multiaddr/pull/55#discussion_r173337049

Comment (ya7ya) [reply src/index.js]: @diasdavid hey, 

**solution 1** : What if we add a `name` property to the `WebSockets` class which isn't gonna be mangled by uglify like so
```javascript
class WebSockets {
 constructor (opts) {
   this.name = 'WebSockets'
 }
}
```

**Solution 2** : use both `instanceOf` and `constructor,name` like so
` if (transport.constructor.name === 'WebSockets' || transport instanceOf WebSockets)`

What do you think.??

Comment (diasdavid) [reply src/index.js]: solution 1 is closer to a real solution. Even better would be just to have a `isWebSockets` method

**Comment by diasdavid** [comment]
See @pgte proposal https://github.com/ipfs/js-ipfs/issues/1131#issuecomment-373299739

**Comment by fsdiogo** [comment]
PR https://github.com/libp2p/js-libp2p/pull/182 fixes this.

**Comment by diasdavid** [comment]
This has been fixed in all the js-ipfs deps thanks to @fsdiogo. @ya7ya thanks for helping us diagnose and provide solutions. Check https://github.com/ipfs/js-ipfs/issues/1321 for more info.

**Pull request closed without merging**